ui/app/src/components/DeleteDialog/DeleteDialogContainer.tsx

```diff
@@ -59,6 +61,12 @@ export function DeleteDialogContainer(props: DeleteDialogContainerProps) {
   const [submitDisabled, setSubmitDisabled] = useState(true);
   const [confirmChecked, setConfirmChecked] = useState(false);
   const authoringBase = useSelection((state) => state.env.authoringBase);
+  const itemsByPath = useItemsByPath();
+  useFetchSandboxItems(dependentItems ?? []);
+  const disabledDependentItems = createPresenceTable(
+    dependentItems?.filter((item) => !itemsByPath[item].availableActionsMap.delete) ?? [],
```

Comment:
Specially with the new logic, is there a moment where `itemsByPath` does't have all the paths loaded? You should throttle for testing
```suggestion
    dependentItems?.filter((path) => !itemsByPath[path]?.availableActionsMap.delete) ?? [],
```